MIGRATION_GUIDE.md

```diff
@@ -37,27 +33,42 @@ resource "snowflake_stream" "stream" {
 to
 
 ```terraform
-resource "snowflake_stream_on_table" "stream" {
+resource "snowflake_stream_on_directory_table" "stream" {
   name     = "stream"
   schema   = "schema"
   database = "database"
 
-  table             = snowflake_table.table.fully_qualified_name
-  append_only       = "true"
+  stage             = snowflake_stage.stage.fully_qualified_name
 
   comment = "A stream."
 }
 ```
 
+Then, follow our [Resource migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/technical-documentation/resource_migration.md).
 
-To use the new `stream_on_directory_table`, change the old `stream` from
+
+## v0.96.0 ➞ v0.97.0
+
+### *(new feature)* snowflake_stream_on_table, snowflake_stream_on_external_table resource
+
+To enhance clarity and functionality, the new resources `snowflake_stream_on_table`, `snowflake_stream_on_external_table` and `snowflake_stream_on_directory_table` have been introduced to replace the previous `snowflake_stream`. Recognizing that the old resource carried multiple responsibilities within a single entity, we opted to divide it into more specialized resources.
+The newly introduced resources are aligned with the latest Snowflake documentation at the time of implementation, and adhere to our [new conventions](#general-changes).
+This segregation was based on the object on which the stream is created. The mapping between SQL statements and the resources is the following:
+- `ON TABLE <table_name>` -> `snowflake_stream_on_table`
+- `ON EXTERNAL TABLE <external_table_name>` -> `snowflake_stream_on_external_table` (this was previously not supported)
+- `ON STAGE <stage_name>` -> `snowflake_stream_on_directory_table`
```

Comment:
we should not list it in v097 changes, its enough to have it in v098. Also, in v098 version migration you don't have to repeat this whole paragraph and other resources, it's enough to add a reference and just describe the newly introduced resource

Reply:
Fixed.